server/dumpbagserver/forms.py

```diff
@@ -0,0 +1,27 @@
+from dumpbagserver import app_config
+from wtforms import Form, StringField
+from wtforms.validators import DataRequired, Length, Regexp, ValidationError
+
+from .bagger import Bagger
+
+
+class SearchForm(Form):
+    search = StringField(
+        "Search",
+        validators=[
+            DataRequired(),
+            Regexp(r"^\w+$", message="Only letters numbers or _."),
+            Length(min=2, max=64),
+        ],
+    )
+
+    def validate_search(form, field):
+        bagger = Bagger(app_config)
+        dumps = bagger.list_dumps()
+        parse = form.search.data
+        find = False
+        for bds in dumps:
+            if parse in bds:
+                find = True
+        if find is False:
+            raise ValidationError(parse + " not in any of the db names.")
```

Comment:
Is it necessary to raise an error rather than returning an empty list? There is code duplication here with what's done in the `/` route and it seems unnecessary at a first glance.
And even if we want an error, I think a validator is not the right place to do it, the validator validates the "input", not the output, if we want an error, it should be raised by the endpoint `/`.

Reply:
This `validate_search` method is to remove: if you want to raise an error, do it in the controller's endpoint.
You should not do the logic of the endpoint in the validor, only validates if the user input is valid.